docs/architecture.md

```diff
@@ -150,6 +150,21 @@ if any write failed to one of the replicas, multiple differing chunk objects
 will be created in the backing store. See [Querier](#querier) for how data is
 deduplicated.
 
+#### Timestamp Ordering
+
+In general, all lines pushed to Loki for a given stream must have a newer
+timestamp than the line received before it. There are, however, two exceptions
+to this rule:
```

Comment:
`There are, however, two cases for handling logs for the same stream with identical nanosecond timestamps:`